File: src/my_cpp_pkg/src/reset_counter.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/set_bool.hpp"

using namespace rclcpp;
using namespace std;

class ResetCounterNode : public Node 
{
public:
    ResetCounterNode() : Node("reset_counter") 
    {
        threads_.push_back(thread(bind(&ResetCounterNode::call_reset_counter,this)));
    }

private:
    void call_reset_counter()
    {
        client_ = this->create_client<example_interfaces::srv::SetBool>("reset_counter");
        while (!client_->wait_for_service(chrono::seconds(1)))
        {
            RCLCPP_INFO(this->get_logger(),"Waiting for server number counter.");
        }
        auto request = std::make_shared<example_interfaces::srv::SetBool::Request>();
        request->data = true;

        auto future = client_->async_send_request(request);
        try
        {
            auto response = future.get();
            if(response->success == true)
            {
                RCLCPP_INFO(this->get_logger(),"Counter has been reset.");
            } 
            else 
            {
                RCLCPP_INFO(this->get_logger(),"Counter hasn't been reset.");
            }
        }
        catch(const std::exception& e)
        {
            RCLCPP_ERROR(this->get_logger(),"Service call failed.");
        }
        
    }
    Client<example_interfaces::srv::SetBool>::SharedPtr client_;
    vector<thread> threads_; 
};

int main(int argc, char **argv)
{
    init(argc, argv);
    auto node = std::make_shared<ResetCounterNode>(); 
    spin(node);
    shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/reset_counter.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import SetBool

class ResetCounterNode(Node): 
    def __init__(self):
        super().__init__("reset_counter") 
        self.call_reset_counter()
    
    def call_reset_counter(self):
        client_ = self.create_client(SetBool,"reset_counter")
        while not client_.wait_for_service(1):
            self.get_logger().warn("Waiting for server number counter.")
        request = SetBool.Request()
        request.data = True

        future = client_.call_async(request)
        # add callback_add_two_ints to be executed when client.call_async(request) is done
        future.add_done_callback(self.callback_reset_counter)   #nho gan gia tri
    
    def callback_reset_counter(self,future):
        try:
            response = future.result()
            if(response.success == True):
                self.get_logger().info("Counter has been reset.")
            else:
                self.get_logger().info("Counter hasn't been reset.")
        except Exception as e:
            self.get_logger().error("Service call failed %r" %(e,))

def main(args=None):
    rclpy.init(args=args)
    node = ResetCounterNode() 
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()